### PR TITLE
test(date-range): add accessibility tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour
- Add `accessibility` Cypress tests for the `Date Range` component to use the new cypress-component-react framework for testing.

### Current behaviour
Old approach for accessibility tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there are newly added tests
- [x] Check if the `date-range.test.js` file passed
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [x] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `Date Range` tests are not running twice.

### CODE REVIEW

**Review by sianford — DISMISSED** [review]

File: src/components/date-range/date-range.test.js [hunk 1]
```diff
@@ -445,4 +486,231 @@ context("Test for DateRange component", () => {
...
+
+    it("should check accessibility with the endInfo as boolean", () => {
+      CypressMountWithProviders(<DateRangeCustom endInfo />);
+
+      cy.checkAccessibility();
+    });
+
+    it.each([
+      [true, "top"],
+      [false, "bottom"],
```
Comment (stephenogorman) [inline src/components/date-range/date-range.test.js:655]: The second parameter (top/bottom) isn't used in this test.

File: src/components/date-range/date-range.test.js [hunk 2]
```diff
@@ -445,4 +486,231 @@ context("Test for DateRange component", () => {
...
+          />
+        );
+
+        icon().realHover();
+        getDataElementByValue("tooltip").should(
+          "have.attr",
+          "data-placement",
+          position
+        );
+        cyRoot().realHover({ position: "topRight" });
```
Comment (stephenogorman) [inline src/components/date-range/date-range.test.js:706]: Looks like this test missing 'cy.checkAccessibility()'

File: src/components/date-range/date-range.test.js (hunk 1 above)
Comment (stephenogorman) [reply src/components/date-range/date-range.test.js:655]: There's only one %s in the description, so that's picking up the first param!

**Review by stephenogorman — APPROVED** [review]

**Review by sianford — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**